pkg/raft/testdata/lagging_commit.txt

```diff
@@ -126,48 +126,48 @@ tick-heartbeat 1
 ----
 ok
 
-# However, the leader does not push the real commit index to the follower 3. It
-# cuts the commit index at the Progress.Match mark, because it thinks that it is
-# unsafe to send a commit index higher than that.
+# The leader knows that the follower 3 is lagging behind, so it sends a MsgApp
+# to fix that.
 process-ready 1
 ----
 Ready MustSync=false:
 Messages:
 1->2 MsgHeartbeat Term:1 Log:0/0
 1->3 MsgHeartbeat Term:1 Log:0/0
-
-# Since the heartbeat message does not bump the follower's commit index, it will
-# take another roundtrip with the leader to update it. As such, the total time
-# it takes for the follower to learn the commit index is:
-#
-#   delay = HeartbeatInterval + 3/2 * RTT
-#
-# This is suboptimal. It could have taken HeartbeatInterval + 1/2 * RTT, if the
-# leader sent the up-to-date commit index in the heartbeat message.
-#
-# See https://github.com/etcd-io/raft/issues/138 which aims to fix this.
+1->3 MsgApp Term:1 Log:1/13 Commit:13
```

Comment:
Does this PR change the default behaviour somehow, or why are so many data-driven tests affected? Would it be possible to have >= 2 commits:

1. The defaults are intact. The introduction of the new code should have close to zero impact on the tests, and they should still pass.
2. Migrate the tests to also test the new behaviour (e.g. enable the new behaviour in one/few tests at a time).